### PR TITLE
Reduce

### PR DESCRIPTION
**Pull request by irkjero**
Add:
- ReduceMin
- ReduceSum
- ReduceProd
- ReduceMean

### CODE REVIEW

File: onnx2torch/node_converters/reduce.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+_TORCH_FUNCTION_FROM_ONNX_TYPE = {
+    'ReduceMax': torch.max,
+    'ReduceMean': torch.mean,
+    'ReduceSum': torch.sum,
+    'ReduceMin': torch.min,
+    'ReduceProd': torch.prod,
+}
+
+
+class OnnxReduce(nn.Module):
```
Comment (ivkalgin) [inline onnx2torch/node_converters/reduce.py:24]: ReduceSum must be realized in separate classes (Dynamic/Static version like squeeze)

File: onnx2torch/node_converters/reduce.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from torch import nn
+
+from onnx2torch.common import OperationConverterResult
+from onnx2torch.common import onnx_mapping_from_node
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+
+_TORCH_FUNCTION_FROM_ONNX_TYPE = {
+    'ReduceMax': torch.max,
```
Comment (ivkalgin) [inline onnx2torch/node_converters/reduce.py:16]: we ca add:
* ReduceL1
* ReduceL2
* ReduceLogSum
* ReduceLogSumExp
* ReduceSumSquare

Comment (irkjero) [reply onnx2torch/node_converters/reduce.py:65]: Добавил.

File: onnx2torch/node_converters/reduce.py (hunk 1 above)
Comment (irkjero) [reply onnx2torch/node_converters/reduce.py]: Done.

File: onnx2torch/node_converters/reduce.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+from onnx2torch.common import SkipTorchTracing
+from onnx2torch.common import get_const_value
+from onnx2torch.common import onnx_mapping_from_node
+from onnx2torch.custom_export_to_onnx import CustomExportToOnnx
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+
+
+def _initialize_none_dim(dim: Union[int, Tuple[int, ...]], input_dim: int):
```
Comment (ivkalgin) [inline onnx2torch/node_converters/reduce.py:28]: is it necessary (line 181)?

Comment (ivkalgin) [inline onnx2torch/node_converters/reduce.py:28]: `Optional[Union`

**Review by ivkalgin — APPROVED** [review]

**Pull request merged**